core/src/main/java/org/lflang/federated/extensions/CExtension.java

```diff
@@ -94,8 +94,8 @@ public void initializeTargetConfig(
     // Also, create the RTI C file and the launcher script.
     // Handle target parameters.
     // If the program is federated, then ensure that threading is enabled.
-    federate.targetConfig.threading = true;
-    federate.targetConfig.setByUser.add(TargetProperty.THREADING);
+    federate.targetConfig.singleThreaded = false;
+    federate.targetConfig.setByUser.add(TargetProperty.SINGLE_THREADED);
```

Comment:
I know that this isn't introduced by this PR. But, specifying `single-threaded` on a federated program should, IMO, be an error condition, not be silently overridden.